### ISSUE TITLE
python3: check revdeps of deps fails when it finds additional packages to install

### ISSUE THREAD

**Issue by joebonrichie** [getsolus/eopkg#43]
Python 2
```
$ sudo eopkg it xiphos --dry-run
Updates available, checking reverse dependencies of runtime dependencies for safety.
Following packages will be installed:
biblesync  enchant16  gtkhtml  libwebkit-gtk  mesalib  mesalib-32bit  mesalib-dbginfo  qt5-base  sword  xiphos
Total size of package(s): 216.64 MB
$ sudo eopkg it xiphos --dry-run --ignore-revdeps-of-deps
Following packages will be installed:
biblesync  enchant16  gtkhtml  libwebkit-gtk  sword  xiphos
```

Python 3
```
$ sudo eopkg4-bin it xiphos --dry-run
Updates available, checking reverse dependencies of runtime dependencies for safety.
System error. Program terminated.
Repo item 
   not found
Please use 'eopkg help' for general help.
Use --debug to see a traceback.
$ sudo eopkg4-bin it xiphos --dry-run --ignore-revdeps-of-deps
Following packages will be installed:
biblesync  enchant16  gtkhtml  libwebkit-gtk  sword  xiphos
Total size of package(s): 36.15 MB
```

From the python 2 version we can see that `mesalib  mesalib-32bit  mesalib-dbginfo  qt5-base` are also wanting to be installed but with python 3, adding these dependencies to the DAG, it chokes.

Very, similar to #42.

Traceback
```
Traceback:
  File "/home/ninya/eopkg/./eopkg-cli", line 73, in <module>
    main()
  File "/home/ninya/eopkg/./eopkg-cli", line 70, in main
    cli.run_command()
  File "/home/ninya/eopkg/pisi/cli/pisicli.py", line 131, in run_command
    self.command.run()
  File "/home/ninya/eopkg/pisi/cli/install.py", line 165, in run
    pisi.api.install(packages, ctx.get_option("reinstall") or reinstall)
  File "/home/ninya/eopkg/pisi/api.py", line 67, in wrapper
    ret = func(*__args, **__kw)
          ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ninya/eopkg/pisi/api.py", line 500, in install
    return pisi.operations.install.install_pkg_names(packages, reinstall)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ninya/eopkg/pisi/operations/install.py", line 53, in install_pkg_names
    G_f, order = plan_install_pkg_names(A)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ninya/eopkg/pisi/operations/install.py", line 359, in plan_install_pkg_names
    G_f.add_dep(name, revdep)
  File "/home/ninya/eopkg/pisi/pgraph.py", line 38, in add_dep
    pkg2 = self.packagedb.get_package(depinfo.package)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ninya/eopkg/pisi/db/packagedb.py", line 96, in get_package
    pkg, repo = self.get_package_repo(name, repo)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ninya/eopkg/pisi/db/packagedb.py", line 213, in get_package_repo
    pkg, repo = self.pdb.get_item_repo(name, repo)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ninya/eopkg/pisi/db/itembyrepo.py", line 40, in get_item_repo
    raise Exception(_("Repo item %s not found") % str(item))
    ```

**Comment by livingsilver94** [comment]
Does that help? https://github.com/getsolus/eopkg/commit/b8dcbcfa3560ad880eeca3c3dcfe8f6351d88e60

**Comment by joebonrichie** [comment]
Yes but the dependency list doesn't match py2
```
$ sudo ./eopkg-cli it xiphos --dry-run
[sudo] password for ninya:
Updates available, checking reverse dependencies of runtime dependencies for safety.
Following packages will be installed:
biblesync  enchant16  gtkhtml  libwebkit-gtk  sword  xiphos
$ sudo eopkg it xiphos --dry-run
Updates available, checking reverse dependencies of runtime dependencies for safety.
Following packages will be installed:
biblesync  enchant16  gtkhtml  libwebkit-gtk  mesalib  mesalib-32bit  mesalib-dbginfo  qt5-base  sword  xiphos
```

**Comment by livingsilver94** [comment]
Uh I think there's something I'm missing here. My outputs are the same on my machine:

```text
fabio@fabio-solus ~ $ sudo eopkg it xiphos --dry-run
Updates available, checking reverse dependencies of runtime dependencies for safety.
Following packages will be installed:
biblesync  enchant16  gtkhtml  libwebkit-gtk  minizip-ng  sword  xiphos
Total size of package(s): 36.21 MB

fabio@fabio-solus ~ $ sudo eopkg4 it xiphos --dry-run
Updates available, checking reverse dependencies of runtime dependencies for safety.
Following packages will be installed:
biblesync  enchant16  gtkhtml  libwebkit-gtk  minizip-ng  sword  xiphos
Total size of package(s): 36.21 MB
```

**Comment by malfisya** [comment]
Additional test : 

```
alfisya  ~  sudo eopkg-cli it xiphos --dry-run                                                                                                                        
Following packages will be installed:
biblesync  enchant16  gtkhtml  sword  xiphos
Total size of package(s): 12.58 MB

alfisya  ~  sudo eopkg it xiphos --dry-run                                                                                                                            
Following packages will be installed:
biblesync  enchant16  gtkhtml  sword  xiphos
Total size of package(s): 12.58 MB

alfisya  ~  sudo eopkg4-bin it xiphos --dry-run                                                                                                                       
Following packages will be installed:
biblesync  enchant16  gtkhtml  sword  xiphos
Total size of package(s): 12.58 MB
```

**Comment by joebonrichie** [comment]
Probably the easiest way to test this is to downgrade mesalib to n-1 from https://packages.getsol.us/unstable/m/mesalib/ and then retry and verify it tries to upgrade `mesalib` when installing xiphos.

**Comment by livingsilver94** [comment]
Ack, will try. Thanks for the tip.

**Comment by TraceyC77** [comment]
Neither eopkg nor eopkg4-bin are showing the mesalib dependency for me after downgrading it

## Results before downgrading mesalib

Python 2

```python
sudo eopkg it xiphos --dry-run
Following packages will be installed:
biblesync  gtkhtml  sword  xiphos
Total size of package(s): 12.53 MB
```

Python 3

```python
sudo eopkg4-bin it xiphos --dry-run
Following packages will be installed:
biblesync  gtkhtml  sword  xiphos
Total size of package(s): 12.53 MB
```

## Results after downgrading mesalib
Downgraded to mesalib-24.0.2-225-1-x86_64

Python 2

```python
sudo eopkg it xiphos --dry-run
Updates available, checking reverse dependencies of runtime dependencies for safety.
Following packages will be installed:
biblesync  gtkhtml  sword  xiphos
Total size of package(s): 12.53 MB
```

Python 3

```python
sudo eopkg4-bin it xiphos --dry-run
Updates available, checking reverse dependencies of runtime dependencies for safety.
Following packages will be installed:
biblesync  gtkhtml  sword  xiphos
Total size of package(s): 12.53 MB
```

Both show that mesalib would be included in an update

```bash
~/Downloads ❯❯❯ eopkg lu | rg 'mesa'
mesalib - Mesa is an OpenGL compatible 3D graphics library.
~/Downloads ❯❯❯ eopkg4-bin lu | rg 'mesa'
mesalib - Mesa is an OpenGL compatible 3D graphics library.
```

**Comment by joebonrichie** [comment]
Resolved by b05fba5d572b29330f72387fef1becaaf318e269
